src/storage/mvcc/reader/scanner.rs

```diff
@@ -478,6 +874,11 @@ impl<S: Snapshot> BackwardScanner<S> {
     }
 }
 
+pub enum Scanner<S: Snapshot> {
+    Backward(BackwardScanner<S>),
+    Forward(ForwardScanner<S>),
+}
```

Comment:
For now, you may need to add helper functions like `unwrap_forward` and `unwrap_backward` so that there won't be so many `match` in your modified tests, which looks too verbose. Additionally I would like to know your plan and purpose of moving them together, i.e. what problem you are going to solve so that you want to refactor it. I know that there can be a performance improvement when doing the refactor in some other ways but the current solution doesn't look to solve my issue so that I don't know what it is for.